src/java_bytecode/java_static_initializers.cpp

```diff
@@ -95,21 +202,40 @@ static void create_clinit_wrapper_symbols(
   symbol_tablet &symbol_table,
   synthetic_methods_mapt &synthetic_methods)
 {
-  const irep_idt &already_run_name =
-    clinit_already_run_variable_name(class_name);
-  symbolt already_run_symbol;
-  already_run_symbol.name = already_run_name;
-  already_run_symbol.pretty_name = already_run_name;
-  already_run_symbol.base_name = "clinit_already_run";
-  already_run_symbol.type = bool_typet();
-  already_run_symbol.value = false_exprt();
-  already_run_symbol.is_lvalue = true;
-  already_run_symbol.is_state_var = true;
-  already_run_symbol.is_static_lifetime = true;
-  already_run_symbol.mode = ID_java;
-  bool failed = symbol_table.add(already_run_symbol);
-  INVARIANT(!failed, "clinit-already-run symbol should be fresh");
-
+  exprt zero =
+    from_integer(static_cast<int>(clinit_statest::NOT_INIT), java_short_type());
+
+  // Create two global static synthetic "fields" for the class "id"
+  // these two variables hold the state of the class initialization algorithm
+  // across calls to the clinit_wrapper
+  add_new_symbol(
+    symbol_table,
+    clinit_state_variable_name(class_name),
+    java_int_type(),
+    zero,
+    false,
+    true);
+
+  add_new_symbol(
+    symbol_table,
+    clinit_local_state_variable_name(class_name),
```

Comment:
Rename this variable `_thread_local` rather than just `_local` to clarify we mean thread-locality not that it's a function local

Reply:
done